System/Process/Internals.hs

```diff
@@ -414,6 +429,8 @@ foreign import ccall unsafe "runInteractiveProcess"
         -> Ptr FD
         -> Ptr FD
         -> Ptr FD
+        -> Ptr CGid
```

Comment:
What I've found:

http://pubs.opengroup.org/onlinepubs/009695399/functions/chown.html
http://stackoverflow.com/questions/21370094/uid-t-type-is-signed-or-not

I wouldn't bother with a mask on the flag, I'd either stick with your original pointer approach or use `-1`.